src/utils/uemParser.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: MIT
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/MIT
+ */
+
+export class UEMParser {
+    public static findSObjects(json: Object): Array<string> {
+        const sObjects = UEMParser.findObjectsWithKey(json, 'objectApiName');
+
+        return sObjects;
+    }
+
+    static findObjectsWithKey(
+        nestedJsonBlock: any,
+        keyToMatch: string
+    ): Array<string> {
+        const results: Array<any> = [];
+
+        if (typeof nestedJsonBlock === 'object') {
+            const keys = Object.keys(nestedJsonBlock);
+
+            for (const key in nestedJsonBlock) {
+                const value = nestedJsonBlock[key];
+                if (key === keyToMatch && typeof value === 'string') {
+                    results.push(nestedJsonBlock[key as keyof Object]);
+                } else {
+                    results.push(
+                        ...UEMParser.findObjectsWithKey(
+                            nestedJsonBlock[key as keyof Object],
+                            keyToMatch
+                        )
+                    );
+                }
+            }
+        } else if (Array.isArray(nestedJsonBlock)) {
```

Comment:
I don't see a use case where this is going to happen. A JS/TS array is `typeof object`.

Reply:
Plus, you don't need to special-case arrays from the logic you have here. `for...in` will return the properties of a JSON object, and the indices of an array.  For example:

```js
let myArray = ["One", "Two", "Three"];
for (const arrayKey in myArray) {
  console.log(`${arrayKey}: ${myArray[arrayKey]}`);
}

// Output:
// 0: One
// 1: Two
// 2: Three
```